msm/skill_repo.py

```diff
@@ -45,24 +46,40 @@ def read_file(self, filename):
         with open(join(self.path, filename)) as f:
             return f.read()
 
-    def update(self):
+    def __prepare_repo(self):
         if not exists(dirname(self.path)):
             makedirs(dirname(self.path))
 
-        with git_to_msm_exceptions():
-            if not isdir(self.path):
-                Repo.clone_from(self.url, self.path)
+        if not isdir(self.path):
+            Repo.clone_from(self.url, self.path)
 
-            git = Git(self.path)
-            git.config('remote.origin.url', self.url)
-            git.fetch()
+        git = Git(self.path)
+        git.config('remote.origin.url', self.url)
+        git.fetch()
 
         try:
             git.checkout(self.branch)
             git.reset('origin/' + self.branch, hard=True)
         except GitCommandError:
             raise MsmException('Invalid branch: ' + self.branch)
 
+    def update(self):
+        with git_to_msm_exceptions():
+            try:
+                git = self.__prepare_repo()
```

Comment:
Here and below, `git = ` isn't needed (and nothing is returned from `self.__prepare_repo`

Reply:
Ah I missed that, remnant from my first approach.